libbeat/cmd/instance/beat.go

```diff
@@ -405,6 +413,18 @@ func (b *Beat) configure() error {
 		return fmt.Errorf("error loading config file: %v", err)
 	}
 
+	// We have to initialize the keystore before any unpack or merging the cloud
+	// options.
+	keystoreCfg, _ := cfg.Child("keystore", -1)
+	pathData, _ := cfg.String("path.data", -1)
```

Comment:
I wonder if `path.data` (typically `/usr/share/beatname/data` is the right place for the keystore. Alternatively we could consider `path.config` (typically `/etc/beatname`)?

We should probably follow what ES/KB did, do you know?

Also, I guess we'd want to check the permissions on the file like we do with the YAML config files.

Reply:
Right, `/etc/beatname` make more sense, I haven't checked ES/KB I will verify that and make the change if needed.

We do check for permission https://github.com/elastic/beats/pull/5687/files#diff-33a534765f797732e272dc741db414f4R233 if `StrictPerm`(0600) is True.